src/tokens/TabWallet.tsx

```diff
@@ -1,231 +1,139 @@
-    navigation.setParams({ activeAssetTab: selectedTab })
+  const hideWalletBalances = useSelector(hideWalletBalancesSelector)
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const { decimalSeparator } = getNumberFormatSettings()
+
+  const supportedNetworkIds = getSupportedNetworkIdsForTokenBalances()
+  const tokens = useSelector((state) => sortedTokensWithBalanceSelector(state, supportedNetworkIds))
+  const totalTokenBalanceLocal = useTotalTokenBalance()
+  const balanceDisplay = hideWalletBalances
+    ? `XX${decimalSeparator}XX`
+    : totalTokenBalanceLocal?.toFormat(2)
+
+  if (tokens.length === 0) {
+    return (
+      <View style={styles.zeroStateContainer}>
+        <Image style={styles.image} source={cowSpaceship} />
+        <Text style={styles.title}>{t('assets.noTokensTitle')}</Text>
+        <Text style={styles.body}>{t('assets.noTokensDescription')}</Text>
+        <Button
+          testID="ZeroStateBuyTokens"
+          style={{ ...styles.button, width: '100%' }}
+          type={BtnTypes.PRIMARY}
+          size={BtnSizes.FULL}
+          onPress={() => {
+            navigate(Screens.FiatExchangeCurrencyBottomSheet, { flow: FiatExchangeFlow.CashIn })
+          }}
+          text={t('assets.buyTokens')}
+        />
+      </View>
+    )
   }
 
   return (
-    // Transparency issue on Android present when a fragment is used - Nested Animated.View prevents it
-    <Animated.View>
-      <Animated.View
-        style={[styles.listHeaderContainer, animatedListHeaderStyles]}
-        onLayout={handleMeasureListHeaderHeight}
-      >
-        <View
-          style={[styles.nonStickyHeaderContainer]}
-          onLayout={handleMeasureNonStickyHeaderHeight}
-        >
-          <View style={styles.balanceSection}>
-            <AssetsTokenBalance showInfo={false} />
-          </View>
-          <ActionsCarousel key={'ActionsCarousel'} />
-        </View>
-        <AssetTabBar
-          activeTab={activeTab}
-          onChange={handleChangeActiveView}
-          displayPositions={displayPositions}
-        />
-      </Animated.View>
-
-      <AssetList
-        activeTab={activeTab}
-        listHeaderHeight={listHeaderHeight}
-        handleScroll={handleScroll}
-      />
-      {showClaimRewards && (
-        <Animated.View
-          style={[styles.footerContainer, animatedFooterStyles]}
-          onLayout={handleMeasureListFooterHeight}
-        >
-          <Button
-            type={BtnTypes.SECONDARY}
-            size={BtnSizes.FULL}
-            text={t('assets.claimRewards')}
+    <View style={styles.container} testID="TabWallet">
+      <View style={styles.row}>
+        <Text style={styles.totalBalance} testID={'TotalTokenBalance'}>
+          {!hideWalletBalances && localCurrencySymbol}
+          {balanceDisplay}
+        </Text>
+        <HideBalanceButton hideBalance={hideWalletBalances} />
+      </View>
+      <ScrollView contentContainerStyle={styles.contentContainerStyle}>
```

Comment:
can we reuse AssetList, I believe it uses SectionList and we also have pull to refresh tokens